doc/operation-and-maintenance/Logging.md

```diff
@@ -0,0 +1,307 @@
+
+```erlang
+    %% Existing handlers
+    {handler, shell_log, logger_std_h, #{
+         level => notice, %% was level => all
+         ...
+    },
+    {handler, disk_log, logger_disk_log_h, #{
+          level => notice,
+          ...
+    },
+    ...
+    %% Module log level
+    {module_level, debug, [ejabberd_c2s]},
+    %% New handler
+    {handler, disk_log_c2s, logger_disk_log_h, #{
+         level => debug,
+         config => #{
+           %% Choose destination:
+           file => "{{mongooseim_log_dir}}/ejabberd_c2s.log",
+           %% Common options:
+           type => wrap,
+           max_no_files => 5,
+           max_no_bytes => 2097152,
+           sync_mode_qlen => 2000,
+           drop_mode_qlen => 2000,
+           flush_qlen => 5000,
+           overload_kill_enable => true
+         },
+         formatter => {mongoose_flatlog_formatter, #{
+           map_depth => 3,
+           term_depth => 50
+         }},
+         filters => [
+           %% That filter matches messages from ejabberd_c2s module
+           {module_filter, {fun mongoose_log_filter:filter_module/2, [ejabberd_c2s]}}
+         ]
+    }}
+```
+
+
+# Setting up Kibana
+
+This example setups ElasticSearch and Kibana for development purposes.
+
+Create a network, so filebeat can find ELK:
+
+```bash
+docker network create logging
+```
+
+Run ELK (consult with the container [docs](https://elk-docker.readthedocs.io/) for more options):
+
+```bash
+docker run -d -p 5601:5601 -p 9200:9200 -p 5044:5044 --network logging --name elk sebp/elk:oss-792
+```
+
+Create a volume for logs:
+
+```bash
+docker volume create mongooseim-logs
+```
+
+Run MongooseIM daemon:
+
+```bash
+docker run -d -t -h mongooseim -v mongooseim-logs:/usr/lib/mongooseim/log \
+    --network logging --name mongooseim -p 5222:5222 mongooseim/mongooseim:latest
+```
+
+The next part is based on [Filebeat's docs](https://www.elastic.co/guide/en/beats/filebeat/current/running-on-docker.html).
+
+Setup filebeat (should be called once, that creates indexes in Elasticsearch):
+
+```bash
+docker run --network logging --rm \
+    docker.elastic.co/beats/filebeat-oss:7.9.2 \
+    setup -E setup.kibana.host=elk:5601 \
+          -E output.elasticsearch.hosts='["elk:9200"]'
+```
+
+Create `filebeat.mongooseim.yml` config file:
+
+```yaml
+filebeat.inputs:
+- paths:
+   - /usr/lib/mongooseim/log/mongooseim.json.1
+  input_type: log
+  json.keys_under_root: true
+  json.add_error_key: true
+  json.overwrite_keys: true
+  json.message_key: text
+
+processors:
+  # Keep the original "when" field too, because of microseconds precision
+  - timestamp:
+      field: when
+      layouts:
+        # Date '2006-01-02T15:04:05.999Z' in mongoose format
+        - '2006-01-02T15:04:05.999+00:00'
+      test:
+        - '2020-09-29T11:25:51.925316+00:00'
+```
+
+Create volume for persistent Filebeat data (so, it would not insert log duplicates, if `mongooseim-filebeat` container is recreated):
```

Comment:
```suggestion
Create a volume for persistent Filebeat data (so it would not insert duplicate logs, if `mongooseim-filebeat` container is recreated):
```